tests/test.cpp

```diff
@@ -1,42 +1,167 @@
+	std::ifstream f("Exodia.json");
+	try {
+		Jsonparser f1(f);
+	}
+	catch (int e) {
+		EXPECT_EQ(e, INVALID_VALUE);
+	}
+}
 
+TEST(Szornytest, Lko_test) {
+	EXPECT_EQ(round(gcd(2.4,1.6)),round(0.8));
+	EXPECT_EQ(round(gcd(2.2, 4.4)), round(2.2));
+	EXPECT_EQ(round(gcd(1.3, 1.1)), round(0.1));
+}
 
-int main(int argc, char ** argv) {
-		::testing::InitGoogleTest(&argc, argv);
-		return RUN_ALL_TESTS();
+TEST(Szornytest, parseUnit_test) {
+	std::ifstream f("Sotetvarazslo.json");
+	std::string szoveg = "Sotetvarazslo.json";
+	const char * fajlnev = "Sotetvarazslo.json";
+	Szorny eredmeny("Sotetvarazslo",250,40,2.0);
+	Jsonparser beolvasas1(f);
+	Jsonparser beolvasas2(szoveg);
+	Jsonparser beolvasas3(fajlnev);
+	Szorny s1 = Szorny::parseUnit(beolvasas1);
+	Szorny s2 = Szorny::parseUnit(beolvasas2);
+	Szorny s3 = Szorny::parseUnit(beolvasas3);
+	EXPECT_EQ(s1==s2, true);
+	EXPECT_EQ(s3 == s2, true);
+	EXPECT_EQ(s1==eredmeny,true);
+}
+
+TEST(Szornytest, harcfunction_Kalandorwin_output_test) {
+	std::string vart = "Sotetvarazslo wins. Remaining HP: 332, current level: 4, current experience: 0";
+	Jsonparser f("Hosarkany.json");
+	Jsonparser f1("Sotetvarazslo.json");
+	Szorny s2 = Szorny::parseUnit(f);
+	Kalandor s1(Szorny::parseUnit(f1));
+	Szorny::harc(s1, s2);
+	testing::internal::CaptureStdout();
+	if (s1.getHp() == 0) std::cout << s2.getName() << " wins. Remaining HP: " << s2.getHp();
+	if (s2.getHp() == 0) std::cout << s1.getName() << " wins. Remaining HP: " << s1.getHp() << ", current level: " << s1.getLvl() << ", current experience: " << s1.getXp();
+	std::string output = testing::internal::GetCapturedStdout();
+	EXPECT_EQ(vart, output);
+	}
+
+TEST(Szornytest, harcfunction_Szornywin_output_test) {
+	std::string vart = "Obelisk wins. Remaining HP: 340";
+	Jsonparser f1("Obelisk.json");
+	Jsonparser f("Hosarkany.json");
+	Szorny s2 = Szorny::parseUnit(f1);
+	Kalandor s1(Szorny::parseUnit(f));
+	Szorny::harc(s1, s2);
+	testing::internal::CaptureStdout();
+	if (s1.getHp() == 0) std::cout << s2.getName() << " wins. Remaining HP: " << s2.getHp();
+	if (s2.getHp() == 0) std::cout << s1.getName() << " wins. Remaining HP: " << s1.getHp() << ", current level: " << s1.getLvl() << ", current experience: " << s1.getXp();
+	std::string output = testing::internal::GetCapturedStdout();
+	EXPECT_EQ(vart, output);
 }
+
+TEST(Szornytest,Kalandor_Szorny_different_test) {
+	Jsonparser f1("Obelisk.json");
+	Jsonparser f("Hosarkany.json");
+	Szorny s2 = Szorny::parseUnit(f1);
+	Kalandor s1(Szorny::parseUnit(f));
+	std::string tipusk=typeid(s1).name();
+	std::string tipuss = typeid(s2).name();
+        tipusk.erase(0,1);
+	tipuss.erase(0,1);
+	EXPECT_EQ(tipusk,"Kalandor");
+	EXPECT_EQ(tipuss,"Szorny");
+}
+
+TEST(Szornytest, Levelup_Xpgain_test) {
+	Kalandor s1("Hosarkany", 300, 30, 2.4, 90, 1);
+	Szorny s2("Sotetvarazslo", 250, 40, 2.0);
+	s1.tamad(s2);
+	Kalandor s3("Hosarkany", 330, 33, 2.4, 20, 2);
+	EXPECT_EQ(s3==s1, true);
```

Comment:
itt is lehet ASSERT_TRUE vagy EXPECT_TRUE attól függően, hogy szeretnéd-e, hogy a test fusson tovább miután hibát talált